File: src/NCU.AnnualWorks/ClientApp/src/shared/models/Auth/CurrentUser.ts
export type CurrentUser = {
  id: string;
  name: string;
  email: string;
  isParticipant: boolean;
  isLecturer: boolean;
  isAdmin: boolean;
  isCustom: boolean;
  isEmployee: boolean;
  avatarUrl: string;
};

export default CurrentUser;

File: lib/NCU.AnnualWorks.Core/Models/CurrentUser.cs
using NCU.AnnualWorks.Authentication.OAuth.Core.Models;

namespace NCU.AnnualWorks.Core.Models
{
    public struct CurrentUser
    {
        public long Id { get; set; }
        public bool IsParticipant { get; set; }
        public bool IsLecturer { get; set; }
        public bool IsAdmin { get; set; }
        public bool IsCustom { get; set; }
        public bool IsEmployee { get; set; }
        public OAuthRequest OAuthCredentials { get; set; }
    }
}
